### PR TITLE
Ubuntu/jammy 23.4

### PR DESCRIPTION
**Pull request by aciba90**
## retain-old-groups.patch

c1f69161575033d1ae1f3804786d2a0eae8b9858 brings back the patch `retain-old-groups.patch`, which was inadvertently dropped as part of https://github.com/canonical/cloud-init/pull/4530 .

## Steps to reproduce

```
new_upstream_snapshot.py --commitish 23.4 -b 2045582
# fix version to match 23.4-0ubuntu1~22.04.1 in the last commit
# sync with upstream/ubuntu/jammy-23.3.x:debian/changelog
dch -r -D jammy ''
git commit -m 'releasing cloud-init version 23.4-0ubuntu0~22.04.1' debian/changelog
git tag ubuntu/23.4-0ubuntu1_22.04.1
```

## Tests

- [x] build-package -- -S -d
- [x] quilt push -a && tox -e py3 && quilt pop -a
- [x] sbuild -d jammy ../out/cloud-init_23.4-0ubuntu1~22.04.1.dsc
- [x] No lintian warnings / errors.

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,24 +1,54 @@
-cloud-init (23.3-0ubuntu0~22.04.1) UNRELEASED; urgency=medium
+cloud-init (23.4-0ubuntu0~22.04.1) jammy; urgency=medium
 
   * d/control: add python3-apt as Recommends to read APT config from apt_pkg
-  * d/cloud-init.maintscript: Remove the unused hook-network-manager
-    conffile. (LP: #2027861)
   * d/p/status-do-not-remove-duplicated-data.patch:
     - Revert behavior downstream, leave duplicate data
+  * d/control: Add python3-passlib as needed for testing
```
Comment (TheRealFalcon) [inline debian/changelog:6]: This along with the Akamai and po/templates lines were in 23.3.1 so should be dropped here

File: debian/changelog [hunk 2]
```diff
@@ -1,24 +1,54 @@
-cloud-init (23.3-0ubuntu0~22.04.1) UNRELEASED; urgency=medium
+cloud-init (23.4-0ubuntu0~22.04.1) jammy; urgency=medium
```
Comment (TheRealFalcon) [inline debian/changelog:1]: Same comment as lunar about version number

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
A few changelog issues, but otherwise looks good

**Comment by aciba90** [comment]
Force-push to fix the version and d/changelog.

**Review by TheRealFalcon — APPROVED** [review]

**Review by TheRealFalcon — APPROVED** [review]

**Pull request merged**